qiskit_experiments/library/driven_freq_tuning/coefficients.py

```diff
@@ -0,0 +1,279 @@
+            solutions = np.roots([deriv * coeff for deriv, coeff in zip((3, 2, 1), fit)])
+            inflection_points = solutions[
+                (solutions.imag == 0) & (np.sign(solutions) == np.sign(amp))
+            ]
+            if len(inflection_points) > 0:
+                # When multiple inflection points are found, use the most outer one.
+                # There could be a small inflection point around amp=0,
+                # when the first order term is significant.
+                amp = min([amp, max(inflection_points, key=abs)], key=abs)
+            trim_amps.append(amp)
+        return tuple(self.convert_amp_to_freq(np.asarray(trim_amps)))
+
+    def __str__(self):
+        # Short representation for dataframe
+        return "StarkCoefficients(...)"
+
+    def __eq__(self, other):
+        return all(
+            [
+                self.pos_coef_o1 == other.pos_coef_o1,
+                self.pos_coef_o2 == other.pos_coef_o2,
+                self.pos_coef_o3 == other.pos_coef_o3,
+                self.neg_coef_o1 == other.neg_coef_o1,
+                self.neg_coef_o2 == other.neg_coef_o2,
+                self.neg_coef_o3 == other.neg_coef_o3,
+            ]
+        )
+
+    def __json_encode__(self) -> dict[str, Any]:
+        return {
+            "class": "StarkCoefficients",
+            "data": {
+                "pos_coef_o1": self.pos_coef_o1,
+                "pos_coef_o2": self.pos_coef_o2,
+                "pos_coef_o3": self.pos_coef_o3,
+                "neg_coef_o1": self.neg_coef_o1,
+                "neg_coef_o2": self.neg_coef_o2,
+                "neg_coef_o3": self.neg_coef_o3,
+                "offset": self.offset,
+            },
+        }
+
+    @classmethod
+    def __json_decode__(cls, value: dict[str, Any]) -> "StarkCoefficients":
+        if not value.get("class", None) == "StarkCoefficients":
+            raise ValueError("JSON decoded value for StarkCoefficients is not valid class type.")
+        return StarkCoefficients(**value.get("data", {}))
+
+
+def retrieve_coefficients_from_service(
+    service: IBMExperimentService,
+    backend_name: str,
+    qubit: int,
+) -> StarkCoefficients:
+    """Retrieve StarkCoefficients object from experiment service.
+
+    Args:
+        service: IBM Experiment service instance interfacing with result database.
+        backend_name: Name of target backend.
+        qubit: Index of qubit.
+
+    Returns:
+        StarkCoefficients object.
+
+    Raises:
+        RuntimeError: When stark_coefficients entry doesn't exist in the service.
+    """
+    try:
+        retrieved = service.analysis_results(
+            device_components=[f"Q{qubit}"],
+            result_type="stark_coefficients",
+            backend_name=backend_name,
+            sort_by=["creation_datetime:desc"],
+            json_decoder=ExperimentDecoder,
+            # Returns the latest value only. IBM service returns 10 entries by default.
```

Comment:
I don't think this comment is necessary. We only want one value so limiting to 1 is reasonable. It should be more efficient, though probably negligibly. The old data that could not be deserialized was just what made me realize in testing that we were retrieving more than one result here.

Reply:
yes, 1 is reasonable mainly for efficiency but I think it's worth having some note about edge case (so that someone save time for debug)